cpp/src/decisiontree/batched-levelalgo/builder_base.cuh

```diff
@@ -157,7 +157,13 @@ struct Builder {
     nHistBins = 2 * max_batch * params.n_bins * n_col_blks * nclasses;
     // x2 for mean and mean-of-square
     nPredCounts = max_batch * params.n_bins * n_col_blks;
-    maxNodes = pow(2, (params.max_depth + 1)) - 1;
+    if (params.max_depth < 13) {
```

Comment:
do we support `max_depth == -1`? If yes, we'll have to take care of it here, right?

Reply:
We do not support it as of now. https://github.com/rapidsai/cuml/blob/48a5416c8fd3c8853089ccf0a1b708198e12d5a4/python/cuml/ensemble/randomforestclassifier.pyx#L193

There is even a validity check, which fails for `max_depth = -1`

Reply:
ah.. right. Makes sense. Thanks for confirming